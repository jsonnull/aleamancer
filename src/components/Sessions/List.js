// @flow
import React from 'react'
import styled from 'styled-components'
import Item from './ListItem'
import type { SessionInfo } from 'types'

const EmptyList = styled.div`margin: 2rem 0;`

const List = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-left: -1rem;
  margin-right: -1rem;
`

type Props = {
  sessions: Array<SessionInfo>,
  setSession: Function
}

const FullList = (props: Props) => {
  const sessions = props.sessions

  if (sessions.length === 0) {
    return (
      <EmptyList>
        {
          "Yikes, looks like you're not a member of any games. Want to start one?"
        }
      </EmptyList>
    )
  }

  return (
    <List>
      {sessions.map(session => (
        <Item
          key={session.id}
          isCurrent={false}
          session={session}
          setSession={props.setSession}
        />
      ))}
    </List>
  )
}

export default FullList
