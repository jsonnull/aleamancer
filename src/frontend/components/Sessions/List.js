// @flow
import React from 'react'
import styled from 'styled-components'
import Item from './ListItem'
import type { GetCurrentUserGamesType } from 'frontend/graphql/queries/getCurrentUserGames'

const EmptyList = styled.div`
  margin: 2rem 0;
`

const List = styled.div`
  display: flex;
  flex-wrap: wrap;
  margin-left: -1rem;
  margin-right: -1rem;
`

const Loading = () => <div>Loading...</div>

type Props = {
  loading: boolean,
  currentUser: { games?: Array<GetCurrentUserGamesType> },
  setSession: Function
}

const FullList = (props: Props) => {
  const { loading, currentUser } = props

  if (loading) {
    return <Loading />
  }

  const sessions = currentUser.games

  if (sessions.length === 0) {
    return (
      <EmptyList>
        {
          "Yikes, looks like you're not a member of any games. Want to start one?"
        }
      </EmptyList>
    )
  }

  return (
    <List>
      {sessions.map(session => (
        <Item
          key={session.id}
          name={session.name}
          setSession={() => props.setSession(session.id, session.name)}
        />
      ))}
    </List>
  )
}

export default FullList
