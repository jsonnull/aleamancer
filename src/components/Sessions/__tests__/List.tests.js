// @flow
import React from 'react'
import renderer from 'react-test-renderer'
import List from '../List.js'

describe('Sessions List component', () => {
  it('renders correctly with no sessions', () => {
    const tree = renderer
      .create(<List sessions={[]} setSession={() => {}} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  const sessions = [
    {
      sessionId: 'id',
      meta: {
        name: 'testName'
      }
    }
  ]
  it('renders correctly with sessions', () => {
    const tree = renderer
      .create(<List sessions={sessions} setSession={() => {}} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
