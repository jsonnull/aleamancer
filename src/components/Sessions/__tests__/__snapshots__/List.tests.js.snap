// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sessions List component renders correctly with no sessions 1`] = `
<div
  className="sc-bxivhb edVTVS"
>
  Yikes, looks like you're not a member of any games. Want to start one?
</div>
`;

exports[`Sessions List component renders correctly with sessions 1`] = `
<div
  className="sc-ifAKCX gpvfwi"
>
  <div
    className="sc-htpNat hpQGuX"
    onClick={[Function]}
  >
    <div>
      <div
        className="sc-bwzfXH cIzktr"
      >
        testName
      </div>
    </div>
  </div>
</div>
`;
